### PR TITLE
[Waiting on other PRs] Add the no-extension case to windows program search

### PR DESCRIPTION
**Pull request by marler8997**
Waiting on the status of #2777 and #2782

### CODE REVIEW

File: std/child_process.zig [hunk 1]
```diff
@@ -768,3 +770,37 @@ pub fn createWindowsEnvBlock(allocator: *mem.Allocator, env_map: *const BufMap)
...
+    onU bool,
+    pub fn init(t: T, u: U) @This() {
+        return @This() {
+            .t = t,
+            .u = u,
+            .onU = false,
+        };
+    }
+    pub fn next(self: *@This()) @typeOf(T.next).ReturnType {
+        return if (self.onU) self.u.next() else self.t.next();
```
Comment (daurnimator) [inline std/child_process.zig]: When do you set `.onU` to `true`?

Comment (marler8997) [reply std/child_process.zig]: in another commit :)

File: std/child_process.zig [hunk 2]
```diff
@@ -768,3 +770,47 @@ pub fn createWindowsEnvBlock(allocator: *mem.Allocator, env_map: *const BufMap)
     i += 1;
     return allocator.shrink(result, i);
 }
+
+// TODO: does this already exist somewhere?
+fn Chain(comptime T: type, comptime U: type) type {
```
Comment (marler8997) [inline std/child_process.zig]: Note that this (and the `OneItemIterator`) should either be moved somewhere else before merging, or they need to be removed if there is something else in the standard library that already provides this.

I'm not familiar enough with the standard library to know if these already exist somewhere in some form.

**Comment by marler8997** [comment]
It's not needed since you'll get a compile error. Checking it might give you an opportunity to provide a nicer error message, but the check might not be simple since they don't need to be the same. Right now U.next just needs to be implicitly convertible to T.next.

**Comment by andrewrk** [comment]
Feel free to re-open when this is ready for review

**Pull request closed without merging**